File: sdks/csharp/sdk/src/PogodocApi/Templates/Types/GeneratePresignedGetUrlResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using PogodocApi.Core;

namespace PogodocApi;

[Serializable]
public record GeneratePresignedGetUrlResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    /// <summary>
    /// Presigned URL to get the template
    /// </summary>
    [JsonPropertyName("presignedUrl")]
    public required string PresignedUrl { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/sdk/src/Templates/Types/GeneratePresignedGetUrlResponse.php
<?php

namespace Pogodoc\Templates\Types;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Core\Json\JsonProperty;

class GeneratePresignedGetUrlResponse extends JsonSerializableType
{
    /**
     * @var string $presignedUrl Presigned URL to get the template
     */
    #[JsonProperty('presignedUrl')]
    public string $presignedUrl;

    /**
     * @param array{
     *   presignedUrl: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->presignedUrl = $values['presignedUrl'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
